### PR TITLE
Instance Unification

### PR DESCRIPTION
**Pull request by gabizou**
Proposal for unified isFlowerPot detection
=============

Whilst it is well understood at this point that a well-specified and suitably future-proof API for Minecraft will need to be appropriately decoupled from Minecraft internals to withstand future changes to the underlying platform as effectively as possible, there are some aspects of the underlying platform which are intrinsically immutable and thus safe to propagate up to the API layer.

In many cases it is fully possible to achieve high levels of abstraction within the API, as demonstrated by the recent additions of the Inventory API and Item Data API. However there are still places where abstraction simply does not provide any measurable benefit to the end user, and adhering more faithfully to the internal architecture of the underlying Minecraft platform has no immediate or developer-facing drawbacks.

Where we can achieve some tangible improvement however is in homogenising developer-facing interfaces in order to provide a consistent and intuitive aspect-oriented development paradigm, reinforced by rigid adherance to integrated vertical software development disciplines.

### The Chaos at the Heart of Minecraft

With Minecraft, what you see is not always what you get. Crowd-sourced names for internal API structures can often fall somewhat short of making any sense whatsoever, but still provide valuable insight for newcomers into Minecraft internals, albeit tainted by occasionally apocryphal or sometimes downright nonsensical assignment of meaning to something that the assignor didn't understand or couldn't be bothered to decipher. Nevertheless, these little curios persist in the deobfuscated Minecraft codebase, and their legacy is one of some entertainment and wry humor when the truth of a particular field or method's origins come to light.

The typical ways of dealing with these conflicts between the *status quo* and common sense resolve into two camps:

* Argue about it pretty much forever
* Accept the nonsense and move on

An example of the former is the most hotly contested field of all time `World.isRemote`, sorry I meant `World.isClient` or is that `World.isR...` okay, okay I'll stop. However one of the most notable examples of the latter is the somewhat controversially named `isFlowerPot` method in Minecraft's `Block` class.

This method and its somewhat peculiar naming was the subject of some discussion until a Mojangsta was nailed down and forced to cough up the original name, which turned out to be something like `handlesOwnClone`, however the charm of the original (kind of quirky and entertaining) `isFlowerPot` was just too strong, and the method maintains its apocryphal nomenclature to this day.

### ClassLoading and You - a ~~song of ice and fire~~ soliloquy in five parts

Everyone is aware that Sponge is powered by ~~black magic~~ Mixins, however Class Loading in Java is a complex topic and Mixins are unable to adequately solve every problem we might throw at them. Fundamental to the successful application of Mixins is the core idea that a mixin is used to implement some aspect of the defined API.

`isFlowerPot` thus presents a problem. How can we adequately represent whether or not something is a flowerpot or not if we don't adequately encode this crucial information in the developer-facing portions of the API?

The answer is: define this information explicitly in each interface.

Taking a purely java-based approach to solving this problem alleviates several issues which would otherwise be caused. Whilst we could of course inject the required code at runtime, this rapidly becomes a very expensive operation when we factor in the sheer number of classes involved, and also the potential conflicts it creates with other non-flowerpot-aware code.

The beauty of this extremely [elegant](https://github.com/SpongePowered/SpongeAPI/commits/hotfix/instance-checks?page=16) and efficient implementation approach is that it provides a unified and 100 percent intuitive way of determining whether literally **any** object is or is not a flowerpot at runtime, without introducing complex transformer logic or polluting existing mixins with unrelated code. This allows mixins to focus on their task of implementing literally everything, whilst `isFlowerPot` logic can be handled in pure Java at a fundamental level within both the API and the Sponge implementation.

Implemented by [Sponge#217](https://github.com/SpongePowered/Sponge/pull/217)

### CODE REVIEW

**Comment by Zidane** [comment]
This PR no longer is mergable. Please address.

**Comment by RobertHerhold** [comment]
Poor Travis... RIP

**Comment by Phroa** [comment]
Travis should only check the very last commit.

**Comment by gabizou** [comment]
@Zidane I don't know what you're talking about.

**Comment by gratimax** [comment]
Why is this under a 'hotfix' branch? This is a pretty important and all-encompassing PR. I argue that since the success of the project depends on this PR, we should put it under the 'urgent/instance-checks' branch.

File: src/main/java/org/spongepowered/api/event/SpongeEventFactory.java [hunk 1]
```diff
@@ -238,6 +247,7 @@ public static BlockChangeEvent createBlockChange(Game game, Cause cause, Locatio
         values.put("cause", Optional.fromNullable(cause));
         values.put("block", block);
         values.put("replacementBlock", replacementBlock);
+
```
Comment (gratimax) [inline src/main/java/org/spongepowered/api/event/SpongeEventFactory.java]: These formatting fixes are out of scope of the PR... please remove them

**Comment by modwizcode** [comment]
When did this become a thing also what magic creates 300+ commits?

**Comment by gabizou** [comment]
> When did this become a thing

It became a thing today.


> what magic creates 300+ commits?

Time... Also, not all commits (and their messages) are the same ;)

**Comment by CptSpaceToaster** [comment]
Diff summary: https://gist.github.com/CptSpaceToaster/e9f5aea794d069c0ab92

**Comment by FyberOptic** [comment]
I can't tell you how many times I needed to know when a block was a flower pot.  This is a godsend.

**Comment by Xemiru** [comment]
internal crying

File: src/main/java/org/spongepowered/api/entity/living/animal/HorseColor.java [hunk 2]
```diff
@@ -42,4 +42,10 @@
      */
     String getName();
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DarkArc) [inline src/main/java/org/spongepowered/api/entity/living/animal/HorseColor.java]: I can now finally easily tell if my Horse is colored FlowerPot, excellent feature! lol

File: src/main/java/org/spongepowered/api/block/BlockProperty.java [hunk 3]
```diff
@@ -84,17 +84,52 @@
     // Subinterface markers for vanilla property types
     interface BooleanProperty extends BlockProperty<Boolean> {
 
+        /**
+         * Checks if this is a flowerpot.
+         *
+         * @return Whether this is a flowerpot
+         */
+        boolean isFlowerPot();
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/api/block/BlockProperty.java]: IMO declaring the method in the base interface is enough.

**Comment by ST-DDT** [comment]
I am confused.
The only part of this PR i get is that it has nothing to do with flowerpots, but the flowerpot is a well known term/name.
Can somebody please describe the purpose of that method to me?
And i also have some more general questions.
Why isnt there a usefull interface declaration? If every class is `FlowerPotable`. This will really improve code cleanless and provide a usefull location for additional javadocs. Which are defenitly are needed if you ask me.
Why do our static enum like classes needs flower pots as well?

**Comment by simon816** [comment]
8/10 Not enough flower pots - IGN

File: src/main/java/org/spongepowered/api/Game.java [hunk 4]
```diff
@@ -127,4 +127,10 @@
      */
     MinecraftVersion getMinecraftVersion();
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/Game.java]: There's a problem here.

File: src/main/java/org/spongepowered/api/GameProfile.java [hunk 5]
```diff
@@ -47,4 +47,10 @@
      */
     Optional<User> getUser();
 
+    /**
+     * Checks if this is a flowerpot.
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/GameProfile.java]: This needs to be fixed.

File: src/main/java/org/spongepowered/api/Server.java [hunk 6]
```diff
@@ -220,4 +220,10 @@
      */
     ConsoleSource getConsole();
 
+    /**
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/Server.java]: This won't work.

File: src/main/java/org/spongepowered/api/attribute/AttributeHolder.java [hunk 7]
```diff
@@ -111,4 +111,10 @@
      */
     void removeModifier(AttributeModifier modifier);
 
+    /**
+     * Checks if this is a flowerpot.
+     *
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/attribute/AttributeHolder.java]: This is broken.

File: src/main/java/org/spongepowered/api/block/BlockSnapshot.java [hunk 8]
```diff
@@ -44,4 +44,10 @@
      */
     BlockState getState();
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/block/BlockSnapshot.java]: Typo?

File: src/main/java/org/spongepowered/api/block/ScheduledBlockUpdate.java [hunk 9]
```diff
@@ -70,4 +70,10 @@
      */
     void setPriority(int priority);
 
+    /**
+     * Checks if this is a flowerpot.
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/block/ScheduledBlockUpdate.java]: Needs to be corrected.

File: src/main/java/org/spongepowered/api/block/tile/Note.java [hunk 10]
```diff
@@ -58,4 +58,11 @@
      * @return The transaction result
      */
     TileDataTransactionResult setNoteData(NoteData data);
+
+    /**
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/block/tile/Note.java]: Formatting is wrong.

File: src/main/java/org/spongepowered/api/block/tile/TileEntityTypes.java [hunk 11]
```diff
@@ -52,4 +52,12 @@ private TileEntityTypes() {}
     public static final TileEntityType SIGN = null;
     public static final TileEntityType SKULL = null;
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    public static boolean isFlowerPot() {
+        return false;
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/block/tile/TileEntityTypes.java]: This needs to be approved by whoever is in charge of this part of the API.

File: src/main/java/org/spongepowered/api/block/tile/carrier/TileEntityCarrier.java [hunk 12]
```diff
@@ -95,4 +95,11 @@
 
     @Override
     TileEntityInventory<TileEntityCarrier> getInventory();
+
+    /**
+     * Checks if this is a flowerpot.
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/block/tile/carrier/TileEntityCarrier.java]: I disagree, this won't work in the long term.

File: src/main/java/org/spongepowered/api/block/tile/data/BannerPatternShapes.java [hunk 13]
```diff
@@ -72,4 +72,13 @@
...
     private BannerPatternShapes() {
     }
+
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    public static boolean isFlowerPot() {
+        return false;
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/block/tile/data/BannerPatternShapes.java]: This is a bad choice.

File: src/main/java/org/spongepowered/api/block/tile/data/FurnaceData.java [hunk 14]
```diff
@@ -61,4 +61,10 @@
      */
     void setRemainingCookTime(int time);
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/block/tile/data/FurnaceData.java]: Have you asked on StackOverflow?

File: src/main/java/org/spongepowered/api/effect/particle/ParticleType.java [hunk 15]
```diff
@@ -92,6 +113,13 @@
          */
         float getDefaultNote();
 
+        /**
+         * Checks if this is a flowerpot.
+         *
+         * @return Whether this is a flowerpot
+         */
+        boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/effect/particle/ParticleType.java]: This is a terrible idea.

File: src/main/java/org/spongepowered/api/entity/EntityInteractionTypes.java [hunk 16]
```diff
@@ -57,4 +57,13 @@
     private EntityInteractionTypes() {
 
     }
+
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    public static boolean isFlowerPot() {
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/EntityInteractionTypes.java]: This doesn't belong in this PR.

File: src/main/java/org/spongepowered/api/entity/hanging/LeashHitch.java [hunk 17]
```diff
@@ -29,4 +29,10 @@
  */
 public interface LeashHitch extends Hanging {
 
+    /**
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/hanging/LeashHitch.java]: This needs to be rethought.

File: src/main/java/org/spongepowered/api/entity/living/Ageable.java [hunk 18]
```diff
@@ -86,4 +86,11 @@
      * a baby.
      */
     void setScaleForAge();
+
+    /**
+     * Checks if this is a flowerpot.
+     *
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/living/Ageable.java]: Horrible choice.

File: src/main/java/org/spongepowered/api/entity/living/Sittable.java [hunk 19]
```diff
@@ -43,4 +43,10 @@
      */
     void setSitting(boolean sitting);
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/living/Sittable.java]: In Scala this would look so much better.

File: src/main/java/org/spongepowered/api/entity/living/animal/Ocelot.java [hunk 20]
```diff
@@ -46,4 +46,10 @@
      */
     void setOcelotType(final OcelotType ocelotType);
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/living/animal/Ocelot.java]: To bad we can't use Java 8, this would have been much simpler.

File: src/main/java/org/spongepowered/api/entity/living/monster/Monster.java [hunk 21]
```diff
@@ -32,4 +32,10 @@
  */
 public interface Monster extends Agent {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/living/monster/Monster.java]: I don't like the implications of this.

File: src/main/java/org/spongepowered/api/entity/living/villager/Professions.java [hunk 22]
```diff
@@ -38,4 +38,13 @@
     private Professions() {
     }
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    public static boolean isFlowerPot() {
+        return false;
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/living/villager/Professions.java]: @mbaxter Are we highlighting random people?

File: src/main/java/org/spongepowered/api/entity/player/gamemode/GameModes.java [hunk 23]
```diff
@@ -39,4 +39,13 @@
     private GameModes() {
     }
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    public static boolean isFlowerPot() {
+        return false;
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/player/gamemode/GameModes.java]: Good attempt, but this can be done better.

File: src/main/java/org/spongepowered/api/entity/projectile/Snowball.java [hunk 24]
```diff
@@ -30,4 +30,10 @@
  */
 public interface Snowball extends DamagingProjectile {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/projectile/Snowball.java]: This still needs a lot of work.

File: src/main/java/org/spongepowered/api/entity/projectile/source/ProjectileSource.java [hunk 25]
```diff
@@ -50,4 +50,11 @@
      * @return The projectile instance that was launched
      */
     <T extends Projectile> T launchProjectile(Class<T> projectileClass, Vector3d velocity);
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/projectile/source/ProjectileSource.java]: You should also do this.

File: src/main/java/org/spongepowered/api/entity/vehicle/minecart/MinecartHopper.java [hunk 26]
```diff
@@ -29,4 +29,10 @@
  */
 public interface MinecartHopper extends MinecartContainer {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/entity/vehicle/minecart/MinecartHopper.java]: I'm unsure about this.

File: src/main/java/org/spongepowered/api/event/GameEvent.java [hunk 27]
```diff
@@ -42,4 +42,10 @@
      */
     Game getGame();
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/GameEvent.java]: Too green, needs more red in this diff.

File: src/main/java/org/spongepowered/api/event/attribute/AttributeBaseChangedEvent.java [hunk 28]
```diff
@@ -49,4 +49,10 @@
      */
     double getNewValue();
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/attribute/AttributeBaseChangedEvent.java]: This makes me sick. Who approved this?

File: src/main/java/org/spongepowered/api/event/block/BlockBreakEvent.java [hunk 29]
```diff
@@ -33,4 +33,10 @@
  */
 public interface BlockBreakEvent extends BlockChangeEvent, ExperienceEvent {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/block/BlockBreakEvent.java]: This might fail at runtime.

File: src/main/java/org/spongepowered/api/event/block/BlockInteractEvent.java [hunk 30]
```diff
@@ -30,4 +30,10 @@
  */
 public interface BlockInteractEvent extends BlockEvent {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/block/BlockInteractEvent.java]: Needs better error handling.

File: src/main/java/org/spongepowered/api/event/block/FluidSpreadEvent.java [hunk 31]
```diff
@@ -29,4 +29,10 @@
  */
 public interface FluidSpreadEvent extends BulkBlockEvent {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/block/FluidSpreadEvent.java]: Users will break this in no time at all.

File: src/main/java/org/spongepowered/api/event/block/tile/FurnaceEvent.java [hunk 32]
```diff
@@ -44,4 +44,11 @@
 
     @Override
     FurnaceData getCurrentData();
+
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/block/tile/FurnaceEvent.java]: Java won't allow this. Stupid Java :(

File: src/main/java/org/spongepowered/api/event/cause/Cause.java [hunk 33]
```diff
@@ -97,4 +97,12 @@ public Object getCause() {
         return this.reason;
     }
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    public boolean isFlowerPot() {
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/cause/Cause.java]: This won't port well.

File: src/main/java/org/spongepowered/api/event/entity/EntityAboutToConstructEvent.java [hunk 34]
```diff
@@ -56,4 +56,10 @@
      */
     Location getLocation();
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/EntityAboutToConstructEvent.java]: This is an implementation detail, assume it won't be a problem.

File: src/main/java/org/spongepowered/api/event/entity/EntityCollisionWithBlockEvent.java [hunk 35]
```diff
@@ -34,4 +34,10 @@
  */
 public interface EntityCollisionWithBlockEvent extends EntityCollisionEvent, BlockEvent {
 
+    /**
+     * Checks if this is a flowerpot.
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/EntityCollisionWithBlockEvent.java]: Did you even compile your changes?

File: src/main/java/org/spongepowered/api/event/entity/EntityEquipEvent.java [hunk 36]
```diff
@@ -53,4 +53,10 @@
      */
     int getSlot();
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/EntityEquipEvent.java]: Good idea, bad implementation.

File: src/main/java/org/spongepowered/api/event/entity/EntityHarvestBlockEvent.java [hunk 37]
```diff
@@ -34,4 +34,10 @@
  */
 public interface EntityHarvestBlockEvent extends EntityEvent, BlockHarvestEvent {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/EntityHarvestBlockEvent.java]: Needs more framework.

File: src/main/java/org/spongepowered/api/event/entity/EntityExpireEvent.java [hunk 38]
```diff
@@ -33,4 +33,10 @@
  */
 public interface EntityExpireEvent extends EntityEvent, CauseTracked {
 
+    /**
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/EntityExpireEvent.java]: There are libs that do that for you.

File: src/main/java/org/spongepowered/api/event/entity/EntityInteractEntityEvent.java [hunk 39]
```diff
@@ -37,4 +37,11 @@
      * @return The target entity
      */
     Entity getTargetEntity();
+
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/EntityInteractEntityEvent.java]: We need a unit test for it.

File: src/main/java/org/spongepowered/api/event/entity/EntityMoveEvent.java [hunk 40]
```diff
@@ -32,4 +32,10 @@
  */
 public interface EntityMoveEvent extends EntityDisplaceEvent {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/EntityMoveEvent.java]: You don't have to reinvent the wheel, just use the JDK method.

File: src/main/java/org/spongepowered/api/event/entity/EntityTeleportEvent.java [hunk 41]
```diff
@@ -52,4 +52,10 @@
      */
     void setKeepsVelocity(boolean keepsVelocity);
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/EntityTeleportEvent.java]: This isn't thread safe.

File: src/main/java/org/spongepowered/api/event/entity/living/LivingChangeHealthEvent.java [hunk 42]
```diff
@@ -33,4 +33,10 @@
  */
 public interface LivingChangeHealthEvent extends LivingEvent, EntityChangeHealthEvent {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/living/LivingChangeHealthEvent.java]: This is client only.

File: src/main/java/org/spongepowered/api/event/entity/living/LivingHarvestBlockEvent.java [hunk 43]
```diff
@@ -34,4 +34,10 @@
  */
 public interface LivingHarvestBlockEvent extends LivingEvent, EntityHarvestBlockEvent {
 
+    /**
+     * Checks if this is a flowerpot.
+     *
+     * @return Whether this is a flowerpot
+     */
+    boolean isFlowerPot();
```
Comment (DDoS) [inline src/main/java/org/spongepowered/api/event/entity/living/LivingHarvestBlockEvent.java]: This sucks.

**Comment by DDoS** [comment]
I commented on the changes. Please fix or I'll close tonight.

**Comment by gabizou** [comment]
Happy April Fools!

File: src/main/java/org/spongepowered/api/entity/living/villager/Professions.java (hunk 22 above)
Comment (zml2008) [inline src/main/java/org/spongepowered/api/entity/living/villager/Professions.java]: @useanalias yes we are!
